src/mps/mpo.jl

```diff
@@ -310,15 +327,14 @@ function densityMatrixApplyMPO(A::MPO, psi::MPS; kwargs...)::MPS
 
     all(x->x!=Index(), [siteindex(A, psi, j) for j in 1:n]) || throw(ErrorException("MPS and MPO have different site indices in applyMPO method 'DensityMatrix'"))
     rand_plev = 14741
-
     psi_c = dag(copy(psi))
     A_c   = dag(copy(A))
     prime!(psi_c, rand_plev)
     prime!(A_c, rand_plev)
-
     for j in 1:n-1
         unique_site_ind = setdiff(findinds(A_c[j], "Site"), findindex(psi_c[j], "Site"))[1]
-        A_c[j] = setprime(A_c[j], 1, unique_site_ind)
+        pl = id(unique_site_ind) == id(commonindex(A_c[j], psi_c[j])) ? 1 : 0
```

Comment:
I don't quite get the logic here. What was wrong with the previous version? Can this be more like the C++ version (not in terms of "Site" tags, but instead using `siteIndex` and `uniqueSiteIndex` functions)?

I had been thinking that all of this logic could be replaced with a single function, maybe called `primeLinkInds!(A, psi, plev)` or `primeInternalInds!(A, psi, plev)`, that primes all "internal" indices of the network made by MPO `A` and MPS `psi`, i.e. all of the link indices of `A` and `psi` as well as the common site indices between the two. I think that is a pretty general code pattern that happens a lot using ITensor: you want to do something to all internal indices or external indices of a network made up of a set of ITensors.

I think in the end it is better to do this in terms of "similar" indices instead of a random prime level, since randomizing the id's seems more fitting than randomizing the prime level. My plan for the C++ version had been to write this function in terms of a `simLinkInds!(A_c, psi_c)` function, but the problem was that required a more general version of eigen which wasn't available in C++. 

Anyway, I guess if it works for your use case for now than it is fine, and we can look into other code logic here later.